packages/data-provider/src/utils/fileUtils.ts

```diff
@@ -30,33 +34,39 @@ export const readHeaders = async (file: Express.Multer.File) => {
 
 /**
  * Reads a .tsv file and parse it to a JSON format
- * @param {string} fileName
+ * @param {string} fileName Full filename path of the .tsv file
+ * @param {Schema} schema Schema to parse data with
  * @returns a JSON format objet
  */
-export const tsvToJson = async (fileName: string): Promise<SchemaData> => {
+export const tsvToJson = async (
+	fileName: string,
+	schema: Schema,
+): Promise<{ records: DataRecord[]; errors?: ParseSchemaError[] }> => {
 	const contents = await fsPromises.readFile(fileName, 'utf-8');
 	const arr = parseTsvToJson(contents);
-	return arr;
+	const parseSchemaResult = parse.parseSchemaValues(arr, schema);
+	if (parseSchemaResult.success) {
+		return { records: parseSchemaResult.data.records };
+	}
```

Comment:
this is a function provided by lectern v2.x to parse raw string data into it's corresponding data type provided by the Dictionary schema